centreon/src/Core/Notification/Application/UseCase/AddNotification/AddNotification.php

```diff
@@ -90,7 +90,7 @@ public function __invoke(
             $this->info('Add notification', ['request' => $request]);
 
             $notificationFactory = new NewNotificationFactory($this->readNotificationRepository);
-            $newNotification = $notificationFactory->create($request->name, $request->isActivated);
+            $newNotification = $notificationFactory->create($request->name, $request->isActivated, $request->timeperiodId);
```

Comment:
You should check that timeperiod ID is valid (see ReadTimeperiodRepositoryInterface::exists())
you can add a new method for that in the NotificationValidator class (see example below)

Also, I would move the validator calls before the creation of the new Notification (just before the factory in fact) 

Example for a new method in the validator: 
```
    public function validateTimeperiod(int $timeperiodId): void
    {
        if (false === $this->readTimeperiodRepository->exists($timeperiodId)) {
            $this->error('Time period does not exist', ['timePeriodId' => $timeperiodId]);

            throw NotificationException::invalidId('timeperiod');
        }
    }
```